olp-cpp-sdk-core/include/olp/core/thread/TaskSchedulerFactory.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <olp/core/CoreApi.h>
+#include <olp/core/porting/make_unique.h>
+
+#include "ThreadPoolTaskScheduler.h"
+
+namespace olp {
+namespace thread {
+class TaskScheduler;
+}
+
+namespace thread {
+/**
+ * @brief Factory that creates TaskScheduler instances to be used used for all
+ * delayed operations.
+ */
+class CORE_API TaskSchedulerFactory {
+ public:
+  virtual ~TaskSchedulerFactory() = default;
+
+  /**
+   * @brief Creates an TaskScheduler instance used for all delayed operations.
+   *
+   * @return An instance of TaskScheduler.
+   */
+  static std::unique_ptr<TaskScheduler> Create() {
+    return std::make_unique<ThreadPoolTaskScheduler>();
+  }
+};
+
+}  // namespace thread
+
```

Comment:
remove newline

Reply:
That's a final single new line at the end of the file. It is present among all the files across the project.

Reply:
I meant between namespace braces:

```suggestion
} // namespace thread
} // namespace olp
```